python/ray/tune/suggest/ax.py

```diff
@@ -122,6 +122,12 @@ def __init__(self,
             use_early_stopped_trials=use_early_stopped_trials)
 
         self._ax = ax_client
+
+        if isinstance(space, dict) and space:
```

Comment:
should we have a better check here, to verify that all values within the space is actually `tune` search space objects?

Reply:
Mixing search spaces should work for most algorithms, e.g. for Hyperopt you could use both `tune.xxx` and `hp.xxx` in the same config. This wouldn't work for `Ax` etc though as we don't support converting from Hyperopt to Ax. However, explicitly checking this means that we introduce dependencies - in order to use Ax you would have to have Hyperopt installed to pass the validation.

Alternatively we could also validate the search spaces for each algorithm. This might be tricky, as we have to keep up with changes in library internals and might miss valid configurations. 

In the first instance and I would thus like to leave it as is. Passing tune search spaces directly to search algorithms is not endorsed anyway. We could consider raising a warning here before conversion to indicate that the config should be passed to `tune.run()` instead.

Reply:
Warning sounds good!